src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
+type Shape = 'triangle'| 'circle'| 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
 
+function roundedArea(number: number): number {
+  return Math.floor(number * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const triangleSides = [a, b, c];
+    const checkValidSides = triangleSides.every((side) => side > 0);
```

Comment:
probably we need to check for all figures that initial input arguments are bigger than 0
let's create separate function for this functionality, that will accept params that we'll pass to the figure constructor and will check are they bigger than 0 or no
and will throw an error if no

with this flow you'll be able to call one function for each figure and be sure that input params will be checked, and you don't need to provide separate functionality for this purpose for each class